node/src/components/deploy_acceptor.rs

```diff
@@ -5,41 +5,126 @@ mod tests;
+
+const ARG_TARGET: &str = "target";
+const ARG_SOURCE: &str = "source";
 
-#[derive(Debug, Error)]
+#[derive(Debug, Error, Serialize)]
+#[allow(clippy::enum_variant_names)]
 pub(crate) enum Error {
+    /// An invalid deploy was received from the client.
+    #[error("block chain has no blocks")]
+    InvalidBlockchain,
+
     /// An invalid deploy was received from the client.
     #[error("invalid deploy: {0}")]
-    InvalidDeploy(DeployValidationFailure),
-    /// An invalid account sent a deploy.
-    #[error("invalid account")]
-    InvalidAccount,
+    InvalidDeployConfiguration(DeployConfigurationFailure),
+
+    /// An invalid deploy was received from the client.
+    #[error("deploy parameter failure: {failure} at prestate_hash: {prestate_hash}")]
+    InvalidDeployParameters {
+        prestate_hash: Digest,
+        failure: DeployParameterFailure,
+    },
+}
+
+/// A representation of the way in which a deploy failed validation checks.
+#[derive(Clone, DataSize, Ord, PartialOrd, Eq, PartialEq, Hash, Debug, Error, Serialize)]
+pub enum DeployParameterFailure {
+    /// Account does not exist
+    #[error("Account does not exist")]
+    NonexistentAccount { account_hash: AccountHash },
+    /// Nonexistent contract at hash
+    #[error("Contract at {contract_hash} does not exist")]
+    NonexistentContractAtHash { contract_hash: ContractHash },
+    #[error("Contract named {name} does not exist in Account's NamedKeys")]
+    NonexistentContractAtName { name: String },
+    /// Nonexistent contract entrypoint
+    #[error("Contract does not have {entry_point}")]
+    NonexistentContractEntryPoint { entry_point: String },
+    /// Contract Package does not exist
+    #[error("Contract Package at {contract_package_hash} does not exist")]
+    NonexistentContractPackageAtHash {
+        contract_package_hash: ContractPackageHash,
+    },
+    #[error("ContractPackage named {name} does not exist in Account's NamedKeys")]
+    NonexistentContractPackageAtName { name: String },
+    /// Invalid contract at given version.
+    #[error("Invalid contract at version: {contract_version}")]
+    InvalidContractAtVersion { contract_version: ContractVersion },
+    /// Invalid associated keys
+    #[error("Account authorization invalid")]
+    InvalidAssociatedKeys,
+    /// Insufficient deploy signature weight
+    #[error("Insufficient deploy signature weight")]
+    InsufficientDeploySignatureWeight,
     /// A deploy was sent from account with insufficient balance.
-    #[error("insufficient balance")]
+    #[error("Insufficient balance")]
     InsufficientBalance,
+    /// A deploy was sent from account with an unknown balance.
+    #[error("Unable to determine balance")]
+    UnknownBalance,
+    /// Transfer is not valid for payment code.
+    #[error("Transfer is not valid for payment code")]
+    InvalidPaymentVariant,
+    /// Missing payment amount.
+    #[error("missing payment argument amount")]
```

Comment:
Minor nit, but still... Some of the error messages start with captial letter while others don't.

Reply:
Thank you, they all start with lower case now